server/public/model/config.go

```diff
@@ -4519,6 +4519,63 @@ func (o *Config) Sanitize(pluginManifests []*Manifest) {
 	o.PluginSettings.Sanitize(pluginManifests)
 }
 
+type FilterTag struct {
+	TypeOfTag string
+	Tag       string
+}
+
+type ConfigFilterOptions struct {
+	TagFilters     []FilterTag
+	RemoveMasked   bool
+	RemoveDefaults bool
+}
+
+type GetConfigOptions struct {
+	RemoveMasked   bool
+	RemoveDefaults bool
+}
+
+func FilterConfig(cfg *Config, opts ConfigFilterOptions) (map[string]any, error) {
+	if cfg == nil {
+		return nil, nil
+	}
+
+	defaultCfg := &Config{}
+	defaultCfg.SetDefaults()
+
+	filteredCfg, err := cfg.StringMap()
+	if err != nil {
+		return nil, err
+	}
+
+	filteredDefaultCfg, err := defaultCfg.StringMap()
+	if err != nil {
+		return nil, err
+	}
+
+	for i := range opts.TagFilters {
+		filteredCfg = structToMapFilteredByTag(filteredCfg, opts.TagFilters[i].TypeOfTag, opts.TagFilters[i].Tag)
+		filteredDefaultCfg = structToMapFilteredByTag(defaultCfg, opts.TagFilters[i].TypeOfTag, opts.TagFilters[i].Tag)
+	}
+
+	if opts.RemoveDefaults {
+		filteredCfg = stringMapDiff(filteredCfg, filteredDefaultCfg)
+	}
```

Comment:
No it wasn't intended actually, it just came easier to do so probably. Because `Diff` sounds something new whereas `Remove...` sounds more of working on the actual struct.